### PR TITLE
Prevent network requests related to ephemeral posts in the widgets editor

### PR DESCRIPTION
**Pull request by adamziel**
## Description

This PR follows up the discussion from https://github.com/WordPress/gutenberg/pull/25831:

> Both navigation editor and widgets editor need a top-level entity record to store all the blocks. While post editor may just use the post entity that comes from the API, there is no corresponding concept in the navigation editor or the widgets editors. At the moment both editors use a "fake" post entity record that's never resolved from the API or persisted to the API. This requires carefully dispatching actions to avoid triggering any resolvers. In other words, it's a hack.

The approach taken here is different. Instead of adding a new concept to `core-data`, this PR simply plugs into `api-fetch` and prevents any unwanted requests from happening. While the implementation differs, the approach is inspired by [Drupal's api-fetch.](https://git.drupalcode.org/project/gutenberg/-/blob/8.x-2.x/js/api-fetch.js)

The advantage of this approach over the current approach is that it's resilient. It doesn't require any initialization, it doesn't rely on implementation details of the resolution pipeline (such as faking `startResolution` and `endResolution`), it won't be easily broken by updates to `core-data`. The only hardcoded part is the path string, I'd rather not use the one specified in `defaultEntities -> baseURL` as the data structure there is just another implementation detail of the resolution system.

The same treatment could be applied to the navigation editor which also uses an in-memory post as a convenience wrapper.

## How has this been tested?
1. Go to widgets editor.
1. Open the network panel and watch out for any 404 and 403.
1. Confirm it works as before this PR - widgets are loaded, possible to manipulate, saving works fine.
1. Confirm there weren't any `widget-area` related network error logged.

## Types of changes
Non-breaking change

## Checklist:
- [x] My code is tested.
- [x] My code follows the WordPress code style. 
- [x] My code follows the accessibility standards. 
- [x] My code has proper inline documentation. 
- [x] I've included developer documentation if appropriate. 
- [x] I've updated all React Native files affected by any refactorings/renamings in this PR.

### CODE REVIEW

**Review by draganescu — COMMENTED** [review]
I wonder if we can do better than plopping 
```
if ( nextOptions?.responseBody ) {
	return nextOptions.responseBody;
}
```

like that. Seems random :)

Otherwise the idea to have a way to stop apiFetch via middleware is a good idea.

File: packages/api-fetch/README.md [hunk 1]
```diff
@@ -54,6 +54,10 @@ Unlike `fetch`, the `Promise` return value of `apiFetch` will resolve to the par
 
 Shorthand to be used in place of `body`, accepts an object value to be stringified to JSON.
 
+#### `responseBody` (`object`)
+
+If specified, the default fetch handler will short-circuit and return the value of this option as a response. This is especially useful in middlewares.
+
```
Comment (noisysocks) [inline packages/api-fetch/README.md:60]: Don't think I've ever seen this in a middleware implementation. Why can't the middleware simply _not_ call the next middleware?

File: packages/edit-widgets/src/components/edit-widgets-initializer/index.js [hunk 2]
```diff
@@ -3,26 +3,7 @@
  */
 import Layout from '../layout';
 
-/**
- * WordPress dependencies
- */
-import { useDispatch } from '@wordpress/data';
-import { useEffect, useState } from '@wordpress/element';
-
 function EditWidgetsInitializer( { settings } ) {
```
Comment (noisysocks) [inline packages/edit-widgets/src/components/edit-widgets-initializer/index.js:6]: This component doesn't do anything now. Let's remove.

**Review by noisysocks — COMMENTED** [review]
It's definitely a nicer hack, but it's still a hack 🙂

I'm happy with it for 5.6. Would prefer that we don't add `responseBody` though and instead just don't call `next()`.

File: packages/api-fetch/README.md (hunk 1 above)
Comment (kevin940726) [reply packages/api-fetch/README.md:60]: Agreed. I think it's possible?

**Comment by draganescu** [comment]
Yes I vote for @noisysocks idea and also call the option something that indicates that apiFetch won't request anything. Setting responseBody is weird :)

File: packages/api-fetch/README.md (hunk 1 above)
Comment (adamziel) [reply packages/api-fetch/README.md:60]: > Why can't the middleware simply not call the next middleware?

Good spot Mr socks, we don't have to get to the end of the chain. Updating now.

> Don't think I've ever seen this in a middleware implementation. 

How could have you when it's only proposed now :D

**Comment by adamziel** [comment]
> It's definitely a nicer hack, but it's still a hack 🙂

It is still a hack indeed, multi-entity saving would be the real answer here - let's explore it for 5.7 :-)

**Review by draganescu — APPROVED** [review]
I think this is a good fix, albeit not the complete solution, to the problem we have.

**Pull request merged**